### DISCUSSION TITLE
deploy_lottery.py recent winner mismatch

### DISCUSSION THREAD

**Discussion by toddabraham** [smartcontractkit/full-blockchain-solidity-course-py#89] [Q&A] [answered]
Just deployed on the rinkeby network and have two different results showing on my machine and Etherscan.io.....

### My Machine results

![image](https://user-images.githubusercontent.com/26919212/134686136-b3c9a8ee-45f6-4c08-992b-fb6b46061177.png)

### Etherscan.io results

![image](https://user-images.githubusercontent.com/26919212/134686386-29cf59c3-6624-4421-815a-f2292a94d713.png)

**Comment by tsopanid** [comment, marked as answer]
I had the same issue.
Seems to have to do with the speed of the network.
If you change `time.sleep(60)` to eg `time.sleep(180)` the test should pass.

**Reply by Programmusic** [reply]
I had this issue too and changing it to 180 worked for me :)  

But be aware that you should change the time.sleep() value in both the test and the function to the same value for this to work correctly. 

You should be able then to check on Etherscan as you contract deploys,  first in the logs an event will appear for RequestedRandomnes, keep refreshing until you see this. At this point in the contract, you will still see the recentWinner 0x0000000000000000. After the wait, you can confirm in your console and also now on Etherscan contract that the recent winner is the same as your Metamask wallet address.

**Comment by AlbertoSinigaglia** [comment]
check https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/35

**Reply by toddabraham** [reply]
@AlbertoSinigaglia I read through #35, my code is set exactly as @PatrickAlphaC explained there `contract = Contract.from_abi(
            contract_type._name, contract_address, contract_type.abi`

Can you elaborate how this helps address the announced winner issue above?

**Reply by PatrickAlphaC** [reply]
check https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/89#discussioncomment-1380108
